### DISCUSSION TITLE
SCI is always 0

### DISCUSSION THREAD

**Discussion by oblancox** [green-coding-solutions/green-metrics-tool#462] [Q&A]
Hello,
I am trying to generate the SCI but this value is always 0. For testing I am using the example application "static-algorithm". The command I use to generate the metrics is: 

> $HOME/green-metrics-tool/runner.py --uri $HOME/example-applications/green-software-foundation-sci/static-algorithm --name static-algorithm --skip-unsafe --docker-prune 

This are the logs :
![image](https://github.com/green-coding-berlin/green-metrics-tool/assets/13055521/e00a14a1-3733-4858-8cc4-696dba067609)
And the SCI configuration: 
![image](https://github.com/green-coding-berlin/green-metrics-tool/assets/13055521/413fb2de-49bb-456d-9cae-a088fbb93a27)

The results:
![image](https://github.com/green-coding-berlin/green-metrics-tool/assets/13055521/3a0ac6d9-222d-42f1-a0b9-0cb0a420de32)

![image](https://github.com/green-coding-berlin/green-metrics-tool/assets/13055521/9458f16c-cfbe-4d36-8327-922c6002f772)

What am I doing wrong?

**Comment by oblancox** [comment]
After some more tests I think that the problem is that the SCI value is very low and is being rounded to 0. After changing the value of sci.I from 45 to 4500 it already generates values ​​other than 0, although only in the bagget
![image](https://github.com/green-coding-berlin/green-metrics-tool/assets/13055521/cf31ac2e-6eab-40b5-8869-603837a719d9)

![image](https://github.com/green-coding-berlin/green-metrics-tool/assets/13055521/fc725617-4c3b-4895-a49e-bf8fdc580469)

**Comment by ArneTR** [comment]
Hey @oblancox 

thanks for raising the issue. I saw you resolved it from your side, but I thought I comment it nonetheless, as this could take probably get some UI/UX improvement from our side.

If the value is not available we display "N/A". The reason for it not being available in sub-phases and also in phases like "Boot" is just that it was not implemented. See this issue: https://github.com/green-coding-berlin/green-metrics-tool/issues/451

2. For the display of the 0.00 I see that this is confusing at first.

We decided to make the dimension in the badges dynamic as they typically will be display invidually in the end (ug, ng, mg, g, kg etc.)

However for the web interface we decided to fix it all to grams (g). This is why it displays as 0.00 and does not flip to the next lower dimension (mg).

This keeps the interface consistent and makes it comparable with another view. 

What do you think of that decision? Would you have preferred it to flip to mg automatic and do you think it would be directly understood when looking at another project with a way higher carbon emission that the dimension might have changed?

Also: How is you opinion if the actualy value is show when hovered? So that on hover a small popup appears that says: "0.13 mg" or similar? Did you look out for something similar?

Or maybe a switch that says "Fix all values to g"

Or a mini info box: "If values are 0 they might be lower than 1 g. Please hover for details"?

**Reply by oblancox** [reply]
As you mentioned at the beginning, it can be confusing if a value of 0.0 appears even though the actual value is shown in the badges  list. IMHO the best option is the last one you propose. Keep the value in g, show the mini info box with the mesage, and show the actual value on hovered.

Thank you very much and congratulations for the work you are doing.
